src/actions/signUpWithEmailAndPassword.tsx

```diff
@@ -0,0 +1,32 @@
+// import { History, LocationState } from 'history';
+import { auth } from '../firebase';
+import { ErrorInterface } from '../components/SignupForm/interfaces/SignupFormInterfaces';
+
+interface SignUpWithEmailAndPasswordAndName {
+  email: string;
+  password: string;
```

Comment:
It should be stored in FormInterfaces.tsx as
```
interface Credentials {
email: string;
  password: string;
  name: string;
}
```

then you could merge Credentials with setError and History like this

```
Credentials & {
//   history: History<LocationState>;
  setError: React.Dispatch<React.SetStateAction<ErrorInterface>>
}
```